pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -913,6 +913,15 @@ func (c *Controller) onSystemNamespaceEvent(_, ns *v1.Namespace, ev model.Event)
 		// network changed, rarely happen
 		// refresh pods/endpoints/services
 		c.onNetworkChange()
+
+		// refresh all ambient workloads
+		if features.EnableAmbientControllers && c.ambientIndex != nil {
+			namespaces := c.namespaces.List(metav1.NamespaceAll, klabels.Everything())
+			for _, cns := range namespaces {
```

Comment:
maybe we should put it in `onNetworkChnage`, so meshNetworkConfig update will also update the workload network label

Reply:
Seems better. Updated and also updated a test.